### PR TITLE
feat: add generic GET endpoint for any resource type

### PR DESCRIPTION
**Pull request by JanakaSandaruwan**
## Purpose
https://github.com/openchoreo/openchoreo/pull/1865

## Checklist
- [ ] Tests added or updated (unit, integration, etc.)
- [ ] Samples updated (if applicable)

## PR Summary: Generic GET Endpoint for Any Resource Type

Files changed by top-level folder:
- internal/ — 2 files changed
  - internal/openchoreo-api/handlers/handlers.go (+4 lines)
  - internal/openchoreo-api/handlers/resource_crud.go (+44 lines)
- api/, config/, pkg/, install/, docs/, openapi/, rca-agent/, make/, cmd/, samples/ — 0 files changed

API / CRD surface changes:
- New protected generic GET endpoint:
  - GET /api/v1/namespaces/{namespaceName}/resources/{kind}/{resourceName} → Handler: h.GetResource
  - Resolves GVK via openChoreoGVK(kind) which hardcodes Group=“openchoreo.dev”, Version=“v1alpha1”, Kind={kind}
  - Returns the unstructured resource object (200) on success
  - Error handling: 404 for not found, 400 for unknown/unsupported kind (meta.IsNoMatchError), 500 for other errors
  - Middleware: registered under protected API group using JWT auth + audit middleware (matches existing protected routes)
- Compatibility risk: Medium — broadens public API surface by accepting arbitrary kinds; code includes TODO marking endpoint temporary/for-removal after API redesign.

Tests added/updated:
- Tests added: 0
- Tests present/updated: none for this endpoint
- Critical paths lacking tests:
  - Parameter validation (missing/empty namespaceName, kind, resourceName)
  - Unsupported kind handling (RESTMapper no-match -> 400)
  - Resource not-found -> 404
  - Happy-path retrieval of unstructured resource
  - Authorization enforcement / RBAC checks for arbitrary kinds

Risk hotspots (short reasons):
1. Authorization / RBAC — Medium: generic handler relies entirely on route-group middleware (JWT + audit). No per-kind RBAC checks; may expose resources if authorization is insufficiently granular.
2. API surface expansion — Medium: accepting any kind (even non-OpenChoreo) increases compatibility/maintenance risk and client coupling; TODO indicates instability.
3. Input validation — Low-Medium: no allowlist for kind; unknown kinds are detected via RESTMapper errors but allow surface to be probed.
4. Secret / sensitive data exposure — Medium: handler returns full unstructured object; could expose sensitive fields from arbitrary resources unless response filtering is applied.
5. Test coverage — High: no tests added; production risk for regressions and unverified error/edge cases.
6. Upgrade / removal path — Low: TODO notes endpoint will be removed after redesign but there is no deprecation/versioning guidance for clients.

Other notes (concise):
- New imports: k8s.io/apimachinery/pkg/api/meta used for meta.IsNoMatchError checks.
- Several existing specific resource handlers (ComponentType, Trait, Workflow, ComponentWorkflow) already follow similar unstructured read/update/delete patterns; this generic endpoint follows same GVK construction but without per-kind validation.

### CODE REVIEW

File: internal/openchoreo-api/handlers/resource_crud.go [hunk 1]
```diff
@@ -602,3 +603,43 @@ func (h *Handler) DeleteComponentWorkflowDefinition(w http.ResponseWriter, r *ht
...
+		log.Error("Missing required path parameters", "namespaceName", namespaceName, "kind", kind, "resourceName", resourceName)
+		writeErrorResponse(w, http.StatusBadRequest, "namespaceName, kind and resourceName are required", services.CodeInvalidInput)
+		return
+	}
+
+	gvk := openChoreoGVK(kind)
+	obj, err := h.getResourceByGVK(ctx, gvk, namespaceName, resourceName)
+	if err != nil {
+		if client.IgnoreNotFound(err) == nil {
+			log.Error("Resource not found", "namespace", namespaceName, "kind", kind, "name", resourceName)
```
Comment (yashodgayashan) [inline internal/openchoreo-api/handlers/resource_crud.go]: Shall we add the namespace, kind, name to logger values and then log to reduce repetition?

Comment (JanakaSandaruwan) [reply internal/openchoreo-api/handlers/resource_crud.go]: This is a cherry-pick of https://github.com/openchoreo/openchoreo/pull/1865. I will add this as an improvement to main branch
